sentry_sdk/integrations/threading.py

```diff
@@ -99,7 +86,8 @@ def _run_old_run_func():
                 with sentry_sdk.use_scope(current_scope_to_use):
                     return _run_old_run_func()
         else:
-            return _run_old_run_func()
+            with sentry_sdk.isolation_scope():
```

Comment:
I think the old logic here was actually wrong but kinda worked because of some fallback code in the scope implementation.

Semantically, if we don't explicitly propagate the scopes, we should ALWAYS have created a new isolation scope for the new thread.